components/auth/hooks.ts

```diff
@@ -0,0 +1,100 @@
+import { FirebaseError } from "firebase/app"
+import {
+  AuthProvider,
+  createUserWithEmailAndPassword,
+  sendPasswordResetEmail,
+  signInWithEmailAndPassword,
+  signInWithPopup
+} from "firebase/auth"
+import { useAsyncCallback } from "react-async-hook"
+import { setProfile } from "../db"
+import { auth } from "../firebase"
+
+const errorMessages: Record<string, string | undefined> = {
+  // TODO: do we want to tell users this? this opens us up to user enumeration attacks
+  "auth/email-already-exists": "You already have an account.",
+  // same here
+  "auth/wrong-password": "Your password is wrong.",
```

Comment:
What do you think about this?

Reply:
I think this is fine, if someone really wanted they could still call these endpoints programmatically